Develop

amaiagalvez opened the pull request.

Mikel-IZT reviewed (commented):
Garapen inguruneetan admin@izt.eus erabiltzen dugu defektuzko erabiltzaile bezala.

info@izt.eus baimen guztiak dituen erabiltzailea izaten da eta admin@izt.eus erabiltzaile arrunt bat izango litzakeena.

eneko-izt committed:
Started to implement the creation of  new task within the column

eneko-izt committed:
Create task in a column

eneko-izt committed:
Merge branch 'develop' into A-14331

Commit:
Merge branch 'A-14331' into develop

eneko-izt committed:
In Dashboard, changed task deleted behaviour. Instead of setting active as false, tasks are softdeleted

eneko-izt committed:
Set an active scope for tasks. Created new relationship for columns that return its active tasks ordered by order column

eneko-izt committed:
To hell with HTML gods.  Since I didnt close TextArea exactly after closing symbol but rather in the next line, it resulted in blank spaces set by default

eneko-izt committed:
Show validation errors

eneko-izt committed:
Assign tags to  new tasks

Commit:
reformat

eneko-izt committed:
Changed task factory to randomly set a delete_at value

eneko-izt committed:
Changed label for attriubute referring to input name

eneko-izt committed:
Got nearly working the error msg only showing in the appropriate field

eneko-izt committed:
Fixed text old value not being reloaded when error on  order. Added validation on user id and column id

eneko-izt committed:
Added custom validation for user and tags (required and exist  in tables)

Commit:
Merge branch 'A-14465' into develop

eneko-izt committed:
Added title and alt attributes to buttons and links

Commit:
Merge branch 'A-14463' into develop

eneko-izt committed:
Database changes for handling Roles for users-

eneko-izt committed:
Fixed changes to migration files and seeder

eneko-izt committed:
Added soft delete field to role_user migration. Updated chart adding roles and role_user tables

Commit:
Merge branch 'A-14581' into develop

eneko-izt committed:
Created user control. Added routes. Created user index view

eneko-izt committed:
Merge branch 'A-14592' into develop

eneko-izt committed:
Merge branch 'develop' into A-14592

eneko-izt committed:
Added create and edit functionality to users view

eneko-izt committed:
Added active column to user table

eneko-izt committed:
Added validation when creating or updating users. Added active field to user index and form views

eneko-izt committed:
CRUD operation in user table working. Includes soft delete in role_user table.
Checkboxes used instead of multiselect

eneko-izt committed:
Replaced checkboxes by select multiple.

eneko-izt committed:
Tidied up the code. Removed isAdmin and isUser methods in user model

eneko-izt committed:
Created user policy to show/hide dashboard menus

eneko-izt committed:
Bloody larahell. When authorizing with can in blade model with full path is required

eneko-izt committed:
password and password confirrmation fields in user form

eneko-izt committed:
Use rule for password confirmation

eneko-izt committed:
Create policies for tag and columns models

eneko-izt committed:
Merge branch 'A-14592' into A-14593

eneko-izt committed:
Apply policy base on user role

eneko-izt committed:
Fixed all the issues related to policy: show/hide menus and route control to CRUD for columns, tags, users

eneko-izt committed:
Fixed bug. Couldn't create a user without roles

eneko-izt committed:
Fixed bug. When creating/updating users with no roles

Commit:
Merge branch 'A-14592' into develop

eneko-izt committed:
Merge branch 'develop' into A-14593

Commit:
Merge branch 'A-14593' into develop

eneko-izt committed:
Merged from develop

eneko-izt committed:
Implemented authorization on menu items and related CRUD forms.

Commit:
Merge branch 'A-14594' into develop

eneko-izt committed:
Userd helper function to determine if the current user has an admin role

eneko-izt committed:
Added helper function. Added entry in composer Autoload

eneko-izt committed:
Create tests for Is_Admin function

eneko-izt committed:
Cosmetic change

eneko-izt committed:
Only show tasks belonging to the logged on user

eneko-izt committed:
Creating test to check each users can only see his own tasks.

eneko-izt committed:
Still implementing unit test.

eneko-izt committed:
Re-organized unit test. Moved test to Unit folder rather than Feature. Removed need to query users we created for tests.

Commit:
Merge branch 'A-14714' into develop

eneko-izt committed:
Re-organized unit test. Avoid searching users before each test.

eneko-izt committed:
Merge branch 'develop' into A-14710

eneko-izt committed:
moved pudate/store code to UserService class

eneko-izt committed:
Created a trait to upper the value of a property

eneko-izt committed:
Changed Trait so it returns upper case of name field.

eneko-izt committed:
Role name in upper case is nowhere in the specs

eneko-izt committed:
Modified to  have a single  validation method.

eneko-izt committed:
Changed condition when verifying a user can see their own tasks

Commit:
Merge branch 'A-14710' into develop

eneko-izt committed:
Created modal form using Bootstrap

eneko-izt committed:
Merge branch 'develop' into A-14714

eneko-izt committed:
Removed validation duplicated code and unused reference to namespaces.

eneko-izt committed:
Merge branch 'develop' into A-14711

eneko-izt committed:
Model to edit a task loads  and shows the selected task

Commit:
TODO batzuk

Commit:
Merge branch 'A-14714' into develop

eneko-izt committed:
Modal form loaded with existing values
Cancel button in modal form working.
Submit button calling controller

eneko-izt committed:
Fixed validation issues in new task form and task editing form.

eneko-izt committed:
Oups. Left a changed I used for testing!

eneko-izt committed:
It's all working.
Not sure Tags validation "Required" is needed or not.

eneko-izt committed:
Pending tasks awating answers

eneko-izt committed:
Added user validation. Removed TODOs

eneko-izt committed:
Show error message if user validation fails

eneko-izt committed:
Merge branch 'develop' into A-14711

eneko-izt committed:
Fixed / amended task validation

eneko-izt committed:
Added debugger settings

eneko-izt committed:
Started  unit testing

eneko-izt committed:
Changes in order to allow unit test solely on validation

eneko-izt committed:
Implemented unit testing for task validation

Commit:
TODO batzuk

Commit:
reformat PSR12

eneko-izt committed:
Created new scopeActive function in BasicTrait( renamed)

eneko-izt committed:
Moved helpers.php to new Helpers folder. Removed duplicated code in controllers

eneko-izt committed:
Removed unused references to packages.
Renamed member variables  in unit test. Relocate member variables

eneko-izt committed:
Moved task validation to a newly created task service

eneko-izt committed:
Added indeoendent test functions rather than a single one with all test.

eneko-izt committed:
Fixed validation problem with roles. Show role related error message if it exists.

eneko-izt committed:
Show Role related error message if it exists.

Commit:
Merge branch 'A-14711' into develop

eneko-izt committed:
Removed unused reference to library

eneko-izt committed:
Only user who created a task can delete it

eneko-izt committed:
Implemented test related to deleting tasks

eneko-izt committed:
Created test classes for models. Implemented Column model test

eneko-izt committed:
Added comment to Column model test

eneko-izt committed:
Corrected method name to camelCase style

eneko-izt committed:
Retrieving users from roles should filter soft deleted user roles.
Added usersWithTrashed method

eneko-izt committed:
Added Role model tests

eneko-izt committed:
Task model and tag model tests

eneko-izt committed:
Role model: removed unused methods
User factory: created states
DatabaseSeeder: Classes imported (use of namespaces)

eneko-izt committed:
Removed unused inactive state in user factory

eneko-izt committed:
Added comments

Commit:
Merge branch 'A-14712' into develop

eneko-izt committed:
Merge branch 'develop' into A-14713

eneko-izt committed:
Changed unit tests

eneko-izt committed:
updated db depecrated schema

eneko-izt committed:
Changed unit test for task column

eneko-izt committed:
Changed database stuff

eneko-izt committed:
Added a new route to share a task with a user.
Changed Task controller to dela with sharing tasks with users.
Changed Home controller to pass list of users to a task in a column (Home view).
Changed task blade to add buttons, and user dropdown list.

eneko-izt committed:
(goes altogether with previous commit)
Added a new route to share a task with a user.
Changed Task controller to dela with sharing tasks with users.
Changed Home controller to pass list of users to a task in a column (Home view).
Changed task blade to add buttons, and user dropdown list.

eneko-izt committed:
Changed test to deal with reviewd issues

eneko-izt committed:
Continue implementing sharing tasks.

eneko-izt committed:
Changed reviewed tests

ualonso011 committed:
Merge pull request #4 from eneko-izt/A-14713

A 14713

eneko-izt committed:
Users can view tasks shared with them by other users.
But a user can only share or edit a task if he is the owner of the task (policy check).

eneko-izt committed:
Check policy in Task controller when user updates a task.

eneko-izt committed:
Show the name of the task owner

eneko-izt committed:
Merge branch 'develop' into A-14715

eneko-izt committed:
Fixed return code when authorize policy fails in CRUD operation

eneko-izt committed:
Added new unit test for user and task models since we created new relations

eneko-izt committed:
Changed default values for factories. Changed tests accordingly

eneko-izt committed:
Added tests related to sharing, unsharing and viewing shared tasks

eneko-izt committed:
Removed softdelete in task_user pivot table

eneko-izt committed:
Added a task model method to return only the users we can  share a task with

eneko-izt committed:
Added test for task model shareableUsers method

eneko-izt committed:
Send an email to inform a task has been shared.

eneko-izt committed:
Form to create a new task should be visible or hidden (initial status) when the user clicks in a New Task button

eneko-izt committed:
Only show one New form. Close any other tht was previously opened

eneko-izt committed:
Use boostrap to hide or show the new task form

koldo123 committed:
Merge pull request #6 from eneko-izt/A-14717

A 14717

eneko-izt committed:
Use mailhog instead of mailstrap.
Created a jobs table to queue emails

eneko-izt committed:
Merge branch 'develop' into A-14715

koldo123 committed:
Merge pull request #5 from eneko-izt/A-14715

A 14715

eneko-izt committed:
Fitxategiak igotzeko aldaketak.

eneko-izt committed:
Accordion behaviour for New Task form. When one is open in a column, any other opened in another column should be closed.

eneko-izt committed:
Improve rendering for file upload

eneko-izt committed:
To be able to download a file

eneko-izt committed:
Set the from part of the email used to inform a task has been shared in config file

eneko-izt committed:
Changed the layout of the task in the dashboard

eneko-izt committed:
Tried to improve the layout a little bit as the task dashboard look start to be too busy.

eneko-izt committed:
Layout

eneko-izt committed:
Fixed issues raise by cide review

koldo123 committed:
Merge pull request #7 from eneko-izt/A-14730

A 14730

eneko-izt committed:
Created a service to cache active columns and viewable tasks by  user in a column

eneko-izt committed:
Columns and Tags should be cached (not tasks)

eneko-izt committed:
phpunit coverage ateratzeko

eneko-izt committed:
Merge branch 'develop' into A-14718

eneko-izt committed:
Changed  PHPUnit.xml to installed specs 9.3+

eneko-izt committed:
Code review related changes

koldo123 committed:
Merge pull request #8 from eneko-izt/A-14718

A-14718

eneko-izt committed:
24h Caching was wronly calculated. Fixed.

eneko-izt committed:
- Implemented caching for user viewable tasks.
- Observers on Column and Task models
- task_user table cannot have an observer so implemented it with the closest thing to an observer. That is through events.

eneko-izt committed:
Added observer to Tag model to refresh caching tags.

eneko-izt committed:
Fixed bug in FileUpload request (bug from  task 14730

eneko-izt committed:
Continue implementing task caching.

eneko-izt committed:
Unit test for CacheService

eneko-izt committed:
Finished implementing user task caching and tests

eneko-izt committed:
Use of DataTables for Columns and Tags pages

eneko-izt committed:
Removed the event when attaching/detaching taskusers.
Instead of this, pivot table model created and an observer for that new model

eneko-izt committed:
Fixed test. Changes related to pivot table now having a model

eneko-izt committed:
Changed query to use scopes

eneko-izt committed:
Added tests for newly created scopes:
UsersNotLoggedIn and UsersNotSharingTask

eneko-izt committed:
Changed the policy to delete a task

eneko-izt committed:
Created test for TaskPolicy

eneko-izt committed:
Removed unused use statements

koldo123 committed:
Merge pull request #9 from eneko-izt/A-14719

A 14719

eneko-izt committed:
Merge branch 'develop' into A-15063

eneko-izt committed:
Merge branch 'develop' into A-14720

eneko-izt committed:
Fixed datatable navigation

eneko-izt committed:
Use databable fur users index.

koldo123 committed:
Merge pull request #11 from eneko-izt/A-15063

A-15063

ualonso011 committed:
Merge pull request #10 from eneko-izt/A-14720

A-14720

eneko-izt committed:
Merge branch 'develop' into A-14723

eneko-izt committed:
Removed Bootstrap installation and dependecies.
Replaced by Tailwind

eneko-izt committed:
WIP using Tailwind. Changed Home page

eneko-izt committed:
Fixed column colour

eneko-izt committed:
I have the Columns and tasks inside displayed and looking good with tailwind

eneko-izt committed:
Sorted modal forms with TailWind when creating tasks and editing them

eneko-izt committed:
Changed all pages to use TailWind

eneko-izt committed:
Fixed and removed parts of the  breadscrum since now we are using tailwind and the side bar

eneko-izt committed:
Applied policies to Side Bar

eneko-izt committed:
A flex change fix. Columns were displayed vertically instead of horizontally

eneko-izt committed:
Added the delete task button

eneko-izt committed:
Fixed a few layout details before going for responsiveness

eneko-izt committed:
Removed cause of javascript exception

eneko-izt committed:
Style change

eneko-izt committed:
Created a menu for admin to access all taks.
Created policies to access Dashboard and admin all tasks.
Created route.
Create a dummy index page based on Tags list

eneko-izt committed:
All tasks index show all the required information. However table is not server-side.

eneko-izt committed:
Server-side datatable processing is working. It includes filtering and ordering.

eneko-izt committed:
Revert "Server-side datatable processing is working. It includes filtering and ordering."

This reverts commit b78b517bc3d133ea0e2d6503b16c396b941d3fa1.

eneko-izt committed:
Revert "All tasks index show all the required information. However table is not server-side."

This reverts commit d7942630faafe9efd89f73f64a9ac4f76d271020.

eneko-izt committed:
Revert "Created a menu for admin to access all taks."

This reverts commit 52d86004f1a7e7510745da2d80a71e94f5ccdda9.

eneko-izt committed:
Use Alpine to show modals (new and edit task)

eneko-izt committed:
Share button fix

eneko-izt committed:
Handler error when email couldnt be sent to queue

eneko-izt committed:
Created routemiddleware to setup user language as locale

eneko-izt committed:
Added transactions to  TÇaskController Store and Share methods. Updated feature test

eneko-izt committed:
Added language to user table. Updated factory and seeder accordingly

eneko-izt committed:
Test for language

ualonso011 committed:
Merge pull request #12 from eneko-izt/A-14723

A 14723

eneko-izt committed:
Merge branch 'develop' into A-14722

eneko-izt committed:
Create JSON language files.
Updated texts in views.
Added language to User forms

eneko-izt committed:
Translation text

eneko-izt committed:
Fixed DataTables texts for translation.

eneko-izt committed:
Fix translations errors

eneko-izt committed:
Translation for email subject.
Fixed problem with overlapping single-double-single quotes in button onclick message

eneko-izt committed:
Created class for languages.
Updated UserController to send that list to the view for editing/creating a user

eneko-izt committed:
Tests for laravel auth messages, json messages and task shared email.

eneko-izt committed:
Added French and Spanish translation files

eneko-izt committed:
Set random factory and seeder with Language class random values

eneko-izt committed:
Added language files for Laravel framework (login, validation)

eneko-izt committed:
Fix translation test for all available languages

eneko-izt committed:
Fixed caching bug. When we changed column id in a task we must clear cache for user tasks in old columns.

eneko-izt committed:
Fixed user validation.

eneko-izt committed:
Created new Modal to create/edit a task. There will be a single instance of this form in the Dashboard (viewing columns and tasks)

eneko-izt committed:
Updated test taking into account language fillable attribute.
Language test done using data providers.

eneko-izt committed:
Added missing translation

eneko-izt committed:
Created validation for UserService.
Added autocomplete to email and password in user new form

ualonso011 committed:
Merge pull request #14 from eneko-izt/A-15199

A-15199

eneko-izt committed:
Created constant for languages. Fixed missing transations.

eneko-izt committed:
Removed silly table column head

eneko-izt committed:
Merge branch 'develop' into A-14722

eneko-izt committed:
Missed merge from develop

eneko-izt committed:
Added attributes names to  validation language file

eneko-izt committed:
Merge branch 'develop' into A-15163

eneko-izt committed:
Single modal for all column new task and editing a task.

eneko-izt committed:
Single modal working.

Commit:
Hizkuntzen array-a itzuli

eneko-izt committed:
Language class  has a constant array  so it can be used as a dataprovider for test

eneko-izt committed:
Fixed test. task fields are now not suffixed with column id. This is because we now have a single modal.

ualonso011 committed:
Merge pull request #13 from eneko-izt/A-14722

A-14722

eneko-izt committed:
Merge branch 'develop' into A-15163

eneko-izt committed:
Translated the missing blade part that were not yet in the branch where the translation was done

eneko-izt committed:
Fixed bug related to single modal change. ColumnId is not used to suffix field names anymore.

eneko-izt committed:
Only show server side errors when modal is automatically reloaded due to server side errors

eneko-izt committed:
Created route, view, sidebar button.
Added index method to task controller.

eneko-izt committed:
Added pagination to all tasks datatable

eneko-izt committed:
Server side datatables working

eneko-izt committed:
Hizkuntzak kontuan hartu Ataza guztien zerrendan (datatables)

ualonso011 committed:
Merge pull request #15 from eneko-izt/A-15163

A-15163

eneko-izt committed:
Merge branch 'develop' into A-14721-Bis

eneko-izt committed:
Added middleware to verify policy in route.
Added test for Server side datatables.

eneko-izt committed:
Fixed ordering by tags and sharing users.

eneko-izt committed:
Made the code a bit cleaner

ualonso011 committed:
Merge pull request #17 from eneko-izt/A-14721-Bis

A-14721-Bis
